src/main/java/cz/cvut/fel/aic/simod/ridesharing/peoplefreightsheuristic/PeopleFreightHeuristicSolver.java

```diff
@@ -0,0 +1,497 @@
+            double planCost = planCostProvider.calculatePlanCost(planDiscomfort, planTime);
+
+            DriverPlan newPlan = new DriverPlan(actionsList, planTime, planCost);
+            returnMap.put(allVehicles.get(i), newPlan);
+        }
+
+        return returnMap;
+    }
+
+    /**
+     * calculates status of taxi 'taxiIdx' at given time
+     */
+    private TaxiStatus calcTaxiStatus (int taxiIdx, int curAlgTime)
+    {
+        List<PlanRequestAction> possibleTaxiSchedule = new ArrayList<>(taxiSchedules.get(taxiIdx));
+        possibleTaxiSchedule.sort(new SortActionsByMaxTime());      // TODO: probably no need to sort it
+
+        if (possibleTaxiSchedule.size() == 0)
+        {
+            return taxiStatuses.get(taxiIdx);
+        }
+
+        boolean personOnBoard = false;
+        int curFreightWeight = 0;
+        int taxiMaxCapacity = allVehicles.get(taxiIdx).getMaxParcelsCapacity();
+        int tempTime = 0;
+        for (int i = 0; tempTime <= curAlgTime; i++)
+        {
+            PlanAction action = possibleTaxiSchedule.get(i);
+            if (action instanceof PlanActionPickup)
+            {
+                PlanActionPickup pickAction = (PlanActionPickup) action;
+                // update the temporary time
+                tempTime = pickAction.getMaxTime();
+                PlanComputationRequest pickRequest = pickAction.request;
+                if (pickRequest instanceof PlanComputationRequestFreight)
+                {
+                    // adding the package onBoard
+                    curFreightWeight += ((PlanComputationRequestFreight) pickRequest).getWeight();
+                }
+                else
+                {
+                    // adding the person onBoard
+                    personOnBoard = true;
+                }
+            }
+            else if (action instanceof PlanActionDropoff)
+            {
+                // update the temporary time
+                tempTime = ((PlanActionDropoff) action).getMaxTime();
+                // if person is onBoard  && this is the dropoff action of person
+                if (personOnBoard && ((PlanActionDropoff) action).request instanceof PlanComputationRequestPeople)
+                {
+                    // remove the person from the taxi
+                    personOnBoard = false;
+                }
+                // if person is not onBoard, then remove package "weight" from the taxi
+                else if (!personOnBoard)
+                {
+                    curFreightWeight -= ((PlanComputationRequestFreight) ((PlanActionDropoff) action).request).getWeight();
+                }
+            }
+        }
+        return new TaxiStatus(taxiMaxCapacity - curFreightWeight, personOnBoard);
+    }
+
+    /**
+     * returns sorted list of new taxi schedule (or null if the schedule is not feasible) and duration of this schedule
+     */
+    private ScheduleWithDuration trySchedule(int taxiIndex, DefaultPlanComputationRequest newRequest)
```

Comment:
This method seems very inefficient, but if it is implemented according to the article, I guess we can leave it for now.